lib/src/cache.dart

```diff
@@ -81,7 +82,21 @@ class Cache {
         File('${sourceDir.path}/.packages').existsSync();
   }
 
-  Future<ProcessResult> installDependencies(Package package) async {
+  Future<Process> raceProcess(Future<Process> futureProcess, int timeout) {
+    if (timeout <= 0) {
+      // Nothing to do here
+      return futureProcess;
+    }
+    // Wrap the futureProcess in timeout, and return it when it's ready (or throw if timeout exceeded!)
+    return futureProcess.then((Process p) =>
+        p.exitCode.timeout(Duration(seconds: timeout), onTimeout: () {
+          Process.killPid(p.pid);
+          throw Exception(
+              'Process with PID ${p.pid} took longer than ${timeout}s to complete. Killed.');
```

Comment:
should this be 

    ${timeout} ms

?

Reply:
Milliseconds for this is too small, so in order for people not to pase huge numbers, the timeout is in **seconds** (it's also documented in the `--help` of the `fetch` operation)

Reply:
Ah, k.  Sounds good!